homeassistant/components/sensor/tibber.py

```diff
@@ -70,13 +70,13 @@ def __init__(self, tibber_home):
 
     async def async_update(self):
         """Get the latest data and updates the states."""
-        now = dt_util.utcnow()
+        now = dt_util.now()
         if self._tibber_home.current_price_total and self._last_updated and \
-           self._last_updated.hour == now.hour and self._newest_data_timestamp:
+           self._last_updated.hour == now.hour and self._last_data_timestamp:
```

Comment:
The data change every hour, but we want to retry if no data are available.

Reply:
If the data hasn't changed since last update, making a state update will not write a new state to the state machine. So you could just try to do that without making the check here. Are there any downsides, you think? It would make the code cleaner.

Reply:
I am not sure I understood your last comment. 
But if I do not return here, the `_update_current_price` will be called every minute

Reply:
Yes, and if all values including state and state attributes remain the same a call to do a state update, will not write a new state to the state machine, so will not clog the database.

The only thing that seems to change, while fetched data is the same, is the `time_diff` variable that is computed from the current time of the call. What will happen there?

Reply:
Yes, there will be no state change and no new state to the state machine.
But I though it would be good avoid calling `_update_current_price` each minute, and then also avoid iterate over all price each minute. 
So the return here is only to avoid unnecessary cpu load.

Reply:
👌